src/db.c

```diff
@@ -912,9 +913,17 @@ void keysCommand(client *c) {
     setDeferredArrayLen(c, replylen, numkeys);
 }
 
+#define DEFAULT_SCAN_COMMAND_COUNT 10
+
+/* The SCAN command's default COUNT is 10.
+ * Since it may store keys + values, the
+ * buffer size is roughly 10 * 2 = 20.
+ * Adding a 20% buffer (20 * 1.2) gives 24. */
+#define MAX_SCAN_ARRAY_BUFFER 24
```

Comment:
It is not the max size but the initial alloc size, right?

```suggestion
#define SCAN_ARRAY_INITIAL_ALLOC 24
```

Reply:
Your advice fits better.